test/lit.cfg

```diff
@@ -1339,36 +1338,21 @@ elif run_os == 'linux-androideabi' or run_os == 'linux-android':
     config.target_runtime = "native"
     config.target_swift_autolink_extract = inferSwiftBinary("swift-autolink-extract")
     config.target_sdk_name = "android"
-    android_link_paths_opt = "-L {} -L {} -L {}".format(
-        shell_quote(make_path(
-            config.android_ndk_path, "sources", "cxx-stl", "llvm-libc++",
-            "libs", ndk_platform_tuple)),
+    android_link_paths_opt = "-L {} -L {}".format(
         shell_quote(make_path(
             toolchain_directory, "lib", "gcc", ndk_platform_triple,
             "{}.x".format(config.android_ndk_gcc_version))),
         shell_quote(make_path(
             toolchain_directory, ndk_platform_triple, "lib")))
-    # Since NDK r14 the headers are unified under $NDK_PATH/sysroot, so the -sdk
-    # switch is not enough. Additionally we have to include both the unified
-    # sysroot, and the architecture sysroot.
-    unified_android_include_path = shell_quote(make_path(
-        config.android_ndk_path, "sysroot", "usr", "include"))
-    architecture_android_include_path = shell_quote(make_path(
-        config.android_ndk_path, "sysroot", "usr", "include",
-        ndk_platform_triple))
-    android_include_paths_opt = "-I {} -I {}".format(
-        unified_android_include_path, architecture_android_include_path)
-    # clang can use -isystem, but Swift cannot.
-    android_include_system_paths_opt = "-isystem {} -isystem {}".format(
-        unified_android_include_path, architecture_android_include_path)
+    # Since NDK r19, the headers and libraries are available in a unified
+    # sysroot at $NDK_PATH/toolchains/llvm/prebuilt/$prebuilt_directory/sysroot,
+    # so the -sdk switch can now be used.
     config.target_build_swift = ' '.join([
         config.swiftc,
         '-target', config.variant_triple,
-        '-Xcc', '--sysroot={}'.format(config.variant_sdk),
-        '-Xclang-linker', '--sysroot={}'.format(config.variant_sdk),
+        '-sdk', config.variant_sdk,
```

Comment:
This does not seem to be correct. `config.variant_sdk` is pointing to the NDK sysroot, because its value is taken from `SWIFT_SDK_android_ARCH_aarch64_PATH`, but the `-sdk` parameter for the Swift compiler should point to the Swift SDK (this is the folder that includes things like `/usr/lib/swift/android/aarch64/swiftrt.o`). If not provided, the resource directory is looked from the location of the `swiftc` binary, which was what was happening until now. Only if one distributes a full SDK (with the Android bits and the Swift bits intermixed) one will be able to use `-sdk`.

Reply:
OK, I didn't know you were implicitly setting the Swift resource directory and this `-sdk` flag overrides that. I notice there is a `resource_dir_opt` variable in lit.cfg, should we start using that to pass in the cross-compiled Android stdlib with the `-resource-dir` flag, if it isn't doing that yet? I use [such a config all the time when cross-compiling](https://github.com/apple/swift/pull/33724/files#diff-1a5a064af373f742294e3b312a5c8d0ab52f5aef78052490e6005165472cda48R1321), works well.